packages/core/src/tests/root.test.ts

```diff
@@ -0,0 +1,47 @@
+import {createRemoteRoot} from '../root';
+
+describe('root', () => {
+  it('does not throw error when no allowed components are set', () => {
+    const root = createRemoteRoot(() => {});
+
+    expect(() => {
+      const Card = root.createComponent('Card');
+      root.appendChild(Card);
+
+      const Button = root.createComponent('Button');
+      root.appendChild(Button);
+    }).not.toThrowError();
+
+    expect(root.children).toHaveLength(2);
+  });
+
+  it('does not throw error for allowed components', () => {
+    const components = ['Card', 'Button'];
+    const root = createRemoteRoot(() => {}, {components});
+
+    expect(() => {
+      const Card = root.createComponent('Card');
+      root.appendChild(Card);
+
+      const Button = root.createComponent('Button');
+      root.appendChild(Button);
+    }).not.toThrowError();
+
+    expect(root.children).toHaveLength(2);
+  });
+
+  it('throws error for not allowed components', () => {
+    const components = ['Card'];
+    const root = createRemoteRoot(() => {}, {components});
+
+    expect(() => {
+      const Card = root.createComponent('Card');
+      root.appendChild(Card);
+
+      const Button = root.createComponent('Button');
+      root.appendChild(Button);
+    }).toThrowError();
```

Comment:
I think you should only test one component in each assertion, and only test the method you are asserting on (`createComponent`, in all of these). It doesn't matter for the previous two tests, but in this one it's not clear if the test fails "properly", because the error might get thrown from a method that you do not expect.

Reply:
Fixed!